src/Decimal.php

```diff
@@ -478,19 +468,11 @@ public function round(int $scale = 0, int $roundMode = self::ROUND_HALF_UP): sta
         $exponent = $scale + 1;
 
         $e = bcpow('10', (string)$exponent);
-        switch ($roundMode) {
-            case static::ROUND_FLOOR:
-                $v = bcdiv(bcadd(bcmul($this, $e, 0), $this->isNegative() ? '-9' : '0'), $e, 0);
-
-                break;
-            case static::ROUND_CEIL:
-                $v = bcdiv(bcadd(bcmul($this, $e, 0), $this->isNegative() ? '0' : '9'), $e, 0);
-
-                break;
-            case static::ROUND_HALF_UP:
-            default:
-                $v = bcdiv(bcadd(bcmul($this, $e, 0), $this->isNegative() ? '-5' : '5'), $e, $scale);
-        }
+        $v = match ($roundMode) {
+            static::ROUND_FLOOR => bcdiv(bcadd(bcmul((string)$this, $e, 0), $this->isNegative() ? '-9' : '0'), $e, 0),
+            static::ROUND_CEIL => bcdiv(bcadd(bcmul((string)$this, $e, 0), $this->isNegative() ? '0' : '9'), $e, 0),
+            default => bcdiv(bcadd(bcmul((string)$this, $e, 0), $this->isNegative() ? '-5' : '5'), $e, $scale),
```

Comment:
I find this slightly less readable, especially since the ROUND_HALF_UP is now implicit instead of explicit as path.

Reply:
oh okay we can add following:
```
static::ROUND_HALF_UP, default => bcdiv(bcadd(bcmul((string)$this, $e, 0), $this->isNegative() ? '-5' : '5'), $e, $scale)
```